apps/cart/models.py

```diff
@@ -0,0 +1,101 @@
+from django.db import models
+from django.contrib.auth import get_user_model
+from django.utils import timezone
+
+from apps.product_catalog.models import Product
+
+User = get_user_model()
+
+# Create your models here.
+class ShoppingSession(models.Model):
+    user = models.ForeignKey(User, null=True, on_delete=models.SET_NULL)  # ForeignKey to User model
+    total = models.DecimalField(max_digits=10, decimal_places=2, default=0.00)  # DecimalField for total
+    created_at = models.DateTimeField(auto_now_add=True)
+    modified_at = models.DateTimeField(auto_now=True)
+
+    def save(self, *args, **kwargs):
+        # Set created_at when the user is created for the first time
+        if not self.id:
+            self.created_at = timezone.now()
```

Comment:
```python
created_at = models.DateTimeField(auto_now_add=True)
modified_at = models.DateTimeField(auto_now=True)
```

No need to implement this manually; these will be done by arguments in the filed model.: `auto_now_add`